libass/ass_directwrite.c

```diff
@@ -668,50 +670,49 @@ static void add_font(IDWriteFont *font, IDWriteFontFamily *fontFamily,
 }
 
 /*
- * Scan every system font on the current machine and add it
- * to the libass lookup. Stores the FontPrivate as private data
- * for later memory reading
+ * When a new font name is requested, called to load that font from Windows
  */
-static void scan_fonts(IDWriteFactory *factory,
-                       ASS_FontProvider *provider)
+static void match_fonts(void *priv, ASS_Library *lib,
+                        ASS_FontProvider *provider, char *name)
 {
-    HRESULT hr = S_OK;
-    IDWriteFontCollection *fontCollection = NULL;
+    ProviderPrivate *provider_priv = (ProviderPrivate *)priv;
     IDWriteFont *font = NULL;
-    hr = IDWriteFactory_GetSystemFontCollection(factory, &fontCollection, FALSE);
+    IDWriteFontFamily *fontFamily = NULL;
+    LOGFONTW lf;
+    HRESULT hr = S_OK;
 
-    if (FAILED(hr) || !fontCollection)
+    memset(&lf, 0, sizeof(LOGFONTW));
+    lf.lfEscapement = GM_COMPATIBLE;
```

Comment:
@Apache553 Can you explain this value? Looking at the [`LOGFONTW` docs](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-logfontw), it doesn’t seem that this constant goes in this field.

Reply:
For comparison, VSFilter simply doesn’t set this field (leaves it zero-filled).

Reply:
This is snippet is copied from another project... After I read the MSDN again just now, I have to admit that this assignment is wrong. It seems that it didn't affect getting the font, since our purpose is to get physical font data used by freetype.